### ISSUE TITLE
amount stuck in the exchange bot, please help :)

### ISSUE THREAD

**Issue by errge** [byteball/btc-exchange#2]
Hey, 

On 2017-10-20 around 19:30 UTC I wanted to do a sell Byteball, buy BTC exchange with the following data payment address given by the bot: DVUHSV7NB2MAZ4FXPDWYH6JFBHD3N2MI.

I don't understand completely why, but the chat and the whole system was very slow, and I had to request multiple times to get this inpayment address. Then the bot replied in a burst and this address was the last one, but it also mentioned this address:  QSTCJWGPHOLRMC64B3FQMJ5EKHXVLU3U.

I already paid the 3.23 GBYTE to the DVUH... address in unit wyIECjF1Hjof41+NgsjeZdaEJq5xfteBFGf9cUtTCAw=, but the bot is not reacting to the payment and I'm afraid it's waiting on the QSTC... address.

Can you please take a look?

Checking the order book, I can say that the amount is not processed yet (the amount of the best buy order didn't change since my inpayment).

Cheers!

**Comment by errge** [comment]
Yep, looked at the code, there needs to be a transaction around {check if address exists, generate new address, insert new address}.

Please double check my proposed fix, as I haven't tried running it and it's my first time using this callbackish sql transaction primitive of yours. :)

**Comment by errge** [comment]
Ohh, and the 3.23 GBYTE that is stuck, I would prefer if you send it back to me, instead of pushing through the exchange on a random price.

**Comment by errge** [comment]
Ping.

**Comment by tonyofbyteball** [comment]
Sorry for late reply, I didn't even know this github issue existed.
You are right, there was a race condition.  However, I didn't accept your PR as sql transactions is not the right way to fix it, they are not locks.  I fixed by using locks.
Sent the GB back, including +10% that you should have received from the November distribution.

**Comment by errge** [comment]
@tonyofbyteball Thank you Tony for your kind review and for the refund.  Good luck with Byteball, it's an amazing project!

**Comment by gambitcoin53** [comment]
i got the same problem
i am stuck at payments but no confirmation message was sent by bot

i got a zero bytes

but no btc was transferred to my btc wallet
 
i sell it at this address EMOXFYFZJRV5BM5XPVY5RK3D3A5X3SR7 which was sent by the bot.

the transaction unit was https://explorer.byteball.org/#1349/hPUAT5VhQ3+9PqJqoMX16gdCwTmNX30XUuFkOs=

ihope you can reverse it too, 

may byteball wallet add is : D5J22WLFEAPKAT25LVMM3TJKRRFHRBOT

thanks,

**Comment by arindammroy** [comment]
my bytes also stuck in exchange bot what to do. wanted to sell some bytes
